XiEditor/LineCache.swift

```diff
@@ -76,6 +76,13 @@ class LineCache {
             return nInvalidBefore + lines.count + nInvalidAfter
         }
     }
+    
+    /// A boolean value indicating whether or not the linecache contains any text.
+    /// - Note: An empty line cache will still contain a single empty line, this
+    /// is sent as an update from the core after a new document is created.
+    var isEmpty: Bool {
+        return lines.count == 1 && lines[0]?.text == ""
```

Comment:
A case can be made this should return true if lines.count == 0 as well. Unlikely to happen much in practice, but maybe could happen when startup / file opening is scripted.